lib/game.rb

```diff
@@ -0,0 +1,144 @@
+    gets
+    puts ''
+    puts 'here we go!'
+    puts ''
+    puts ''
+  end
+
+  def create_full_deck
+    values = %w[2 3 4 5 6 7 8 9 10 Jack Queen King Ace]
+    ranks = [2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14]
+    values_and_ranks = values.zip(ranks)
+    values_and_ranks.map do |x|
+      full_deck << [x, :diamond].flatten
+      full_deck << [x, :heart].flatten
+      full_deck << [x, :spade].flatten
+      full_deck << [x, :club].flatten
+    end
+
+    @full_deck = full_deck.map do |x|
+      Card.new(x[2], x[0], x[1])
+    end
+  end
+
+  def shuffle_the_deck
+    @full_deck = @full_deck.shuffle
+  end
+
+  def split_deck
+    @deck1 = Deck.new(full_deck.slice!(0, 26))
+    @deck2 = Deck.new(full_deck)
+  end
+
+  def create_players
+    @player1 = Player.new('Megan', @deck1)
+    @player2 = Player.new('Aurora', @deck2)
+  end
+
+  def game_over?
+    @turn.player1.has_lost? || @turn.player2.has_lost?
+  end
+
+  # def endgame?
+  #   turn.player1.deck.cards.length < 3 || turn.player2.deck.cards.length < 3
+  # end
+
+  # def three_card_endgame?
+  #   turn.player1.deck.cards.length == 3 || turn.player2.deck.cards.length == 3
+  # end
+
+  def two_card_endgame?
+    turn.player1.deck.cards.length == 2 || turn.player2.deck.cards.length == 2
+  end
+
+  def two_card_endgame(counter)
+    case turn.type
+    when :basic
+      normal_game_play(counter)
+    when :war
+      if @player2.deck.cards[2].nil?
+        turn.player1.deck.cards.concat(turn.player2.deck.cards.pop(2))
+      elsif @player1.deck.cards[2].nil?
+        turn.player2.deck.cards.concat(turn.player1.deck.cards.pop(2))
+      end
+
+    else
+      'not allowed'
+    end
+  end
+
+  def one_card_endgame?
+    turn.player1.deck.cards.length == 1 || turn.player2.deck.cards.length == 1
+  end
+
+  def one_card_endgame
+    if turn.player1.deck.cards[0].rank < turn.player2.deck.cards[0].rank
+      turn.player2.deck.cards << turn.player1.deck.cards.pop
+    else
+      turn.player1.deck.cards << turn.player2.deck.cards.pop
+    end
+  end
+
+  def normal_game_play(counter)
+    case turn.type
+    when :mutually_assured_destruction
+      turn.pile_cards
+      p "turn #{counter}: No winner: mutually assured destruction 6 cards removed"
+      p "Megan: #{turn.player1.deck.cards.count}| Aurora: #{turn.player2.deck.cards.count}"
+    when :war
+      turn.pile_cards
+      p "turn #{counter}: WAR -  #{hand_winner.name} won #{turn.spoils_of_war.count} cards"
+      turn.award_spoils(hand_winner)
+      p "Megan: #{turn.player1.deck.cards.count}| Aurora: #{turn.player2.deck.cards.count}"
+    when :basic
+      turn.pile_cards
+      p "turn #{counter}: #{hand_winner.name} won #{turn.spoils_of_war.count} cards"
+      turn.award_spoils(hand_winner)
+      p "Megan: #{turn.player1.deck.cards.count}| Aurora: #{turn.player2.deck.cards.count}"
+    else
+      'uh oh'
+    end
+  end
+
+  def make_turn
```

Comment:
This method is never called anywhere so there is no turn. You probably saw a lot of 'undefined method of nil class' errors where `turn` was involved. We need to create our turn in order to call methods on it.